Cicd uplift

duttonw opened the pull request.

ThrawnCA committed:
Merge pull request #30 from qld-gov-au/develop

Develop to master

duttonw committed:
Merge pull request #34 from qld-gov-au/develop

Develop to master - update for CKAN 2.11+ and add more test reports

duttonw committed:
chore: uplift cicd, pyright types addition

duttonw committed:
chore: pr comment updates

duttonw committed:
chore: pr comment updates

ThrawnCA commented on .github/workflows/pytest.yml:
I am not a fan of this name for a workflow that also includes Behave-based scenario testing.

duttonw replied:
Do you want to split behave testing into its own workflow and in doing so make it run faster?

ThrawnCA replied:
Tests are fast anyway, most of the time is container setup.

ThrawnCA commented on .github/workflows/pytest.yml:
```diff
@@ -0,0 +1,109 @@
+      - name: Upload HTML coverage report
+        uses: actions/upload-artifact@v4
+        if: ${{ !cancelled() }}
+        continue-on-error: true
```
This can conceal step failures when the upload doesn't work. Perhaps it should come after the test summary action?

ThrawnCA commented on README.md:
````diff
@@ -75,9 +91,15 @@ To install this plugin for development:
 1. Install the plugin and its dependencies:
-
+    ```
+    pip install ckanext-ssm-config
````
This won't perform a source install, so the next line may not work.

ThrawnCA replied:
`pyproject.toml` might be a better approach.

ThrawnCA commented on ckanext/ssm/plugin.py:
```diff
@@ -28,16 +37,20 @@ def get_instance_identity_document():
-    implements(IConfigurer, inherit=True)
+    plugins.implements(plugins.IConfigurer, inherit=True)
+
+    client: 'SSMClient|None' = None
```
It would probably be simpler to pass this to the functions as a parameter. We can do the null check in the `update_config` function and then guarantee it's not null when passing it on.

duttonw committed:
chore: improvements from pr comments

ThrawnCA commented on README.md:
````diff
@@ -93,7 +93,7 @@ To install this plugin for development:
     ```
     pip install ckanext-ssm-config
-    pip install -r $SRC_DIR/ckanext-ssm-config/requirements.txt
+    pip install -r $(pip show ckanext-ssm-config | grep Location | awk '{print $2}')/ckanext-ssm-config/requirements.txt
````
This is certainly not industry best practice for installing requirements.

ThrawnCA commented:
Why have you forked the repo instead of just making a branch?

ThrawnCA commented on pyproject.toml:
```diff
@@ -0,0 +1,140 @@
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",]
+keywords = [ "CKAN", "extension", "aws param store", "aws", "param store"]
+dependencies = [ "typing_extensions",]
```
This should include everything from `requirements.txt`

ThrawnCA committed:
cleanup

- Drop incorrect developer installation instructions from README
- Simplify plugin.py to reduce null checks and typing complexity

duttonw committed:
Merge pull request #1 from qld-gov-au/cicd_uplift

cleanup

ThrawnCA committed:
fix unit testing to pass 'client' as a parameter

ThrawnCA committed:
Merge branch 'cicd_uplift' of https://github.com/duttonw/ckanext-ssm-config into cicd_uplift

duttonw committed:
Merge pull request #2 from qld-gov-au/cicd_uplift

Fix unit tests to pass 'client' as a parameter
